### PR TITLE
`Paywalls`: new `PaywallViewControllerDelegate.paywallViewController(_:didChangeSizeTo:)`

### PR DESCRIPTION
**Pull request by vegaro**
- Add `paywallViewControlleSizeDidChange` to notify of size changes
- Added back top anchor to footers for auto layout to size correctly

### CODE REVIEW

File: RevenueCatUI/UIKit/PaywallViewController.swift [hunk 1]
```diff
@@ -139,6 +136,9 @@ public protocol PaywallViewControllerDelegate: AnyObject {
     @objc(paywallViewControllerWasDismissed:)
     optional func paywallViewControllerWasDismissed(_ controller: PaywallViewController)
 
+    @objc(paywallViewControlleSizeDidChange:)
+    optional func paywallViewControlleSizeDidChange(_ size: CGSize)
```
Comment (NachoSoto) [inline RevenueCatUI/UIKit/PaywallViewController.swift:140]: Let's add a doc to this and the sender controller like the others have:
```suggestion
    /// For internal use only.
    @objc(paywallViewControlle:didChangeSize:)
    optional func paywallViewController(_controller: PaywallViewController, didChangeSize size: CGSize)
```

File: RevenueCatUI/UIKit/PaywallViewController.swift [hunk 2]
```diff
@@ -72,6 +72,9 @@ public class PaywallViewController: UIViewController {
                 guard let self = self else { return }
                 self.delegate?.paywallViewController?(self, didFinishRestoringWith: customerInfo)
             }
+            .onSizeChange { [weak self] in
+                self?.delegate?.paywallViewControlleSizeDidChange?($0)
+            }
```
Comment (NachoSoto) [inline RevenueCatUI/UIKit/PaywallViewController.swift:77]: Then this becomes:
```suggestion
            .onSizeChange { [weak self] in
                guard let self = self else { return }
                self.delegate?.paywallViewController(self, didChangeSize: $0)
            }
```

**Review by NachoSoto — COMMENTED** [review]
🎉 

Just a couple of comments from our initial prototype implementations.
Also this needs to go into the API tester: https://github.com/RevenueCat/purchases-ios/blob/main/Tests/APITesters/RevenueCatUIAPITester/SwiftAPITester/PaywallViewControllerAPI.swift#L31

**Comment by NachoSoto** [comment]
Changed the label to `refactor` since this is meant for internal use only.

**Review by tonidero — COMMENTED** [review]
Looks great! After Nacho's comments are fixed will rereview.

File: RevenueCatUI/UIKit/PaywallViewController.swift (hunk 2 above)
Comment (NachoSoto) [reply RevenueCatUI/UIKit/PaywallViewController.swift]: Maybe `didChangeSizeTo`?

**Review by NachoSoto — APPROVED** [review]

**Pull request merged**